ps_customtext.php

```diff
@@ -268,19 +274,54 @@ public function installFixtures()
             ),
         );
 
-        $shops_ids = Shop::getShops(true, null, true);
+        $shopsIds = Shop::getShops(true, null, true);
+        $text = array();
 
         foreach ($tab_texts as $tab) {
-            $info = new CustomText();
             foreach (Language::getLanguages(false) as $lang) {
-                $info->text[$lang['id_lang']] = $tab['text'];
-            }
-            foreach ($shops_ids as $id_shop) {
-                $info->id_shop = $id_shop;
-                $return &= $info->add();
+                $text[$lang['id_lang']] = $tab['text'];
             }
         }
 
+        foreach ($shopsIds as $idShop) {
+            $info = new CustomText();
+            $info->id_info = 1;
+            $info->id_shop = (int)$idShop;
+            $info->text = $text;
+
+            $return &= $info->add();
+        }
+
         return $return;
     }
+
+    /**
+     * Check if $info is the requested object
+     * @param CustomText $info
+     * @param int $idShop
+     * @return bool
+     */
+    private function isRequestedCustomTextObject($info, $idShop)
+    {
+        $infoShop = is_array($info->id_shop) ? reset($info->id_shop) : $info->id_shop;
+
+        return (int)$infoShop === $idShop ? true : false;
+    }
+
+    /**
+     * Add CustomText when adding a new Shop
+     * @param array $params
+     * @throws PrestaShopException
```

Comment:
Are you sure? I don't see an exception thrown here :)

Reply:
That was generated with PhpStorm, probably because the add function throws PrestaShopException.
You are right, I've removed it in order to not violate the [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) principle.

Thanks !